doc/dynamic-data.md

```diff
@@ -1,4 +1,62 @@
-# Referencing Dynamic Data
+# Referencing dynamic data
 
-Sometimes it can be useful to refer to data in a Container Linux Config that
-isn't known until a machine boots, like it's network address. 
+## Overview
+
+Sometimes it can be useful to refer to data in a Container Linux Config that isn't known until a machine boots, like its network address. This can be accomplished with [coreos-metadata][coreos-metadata] by having it run before a given unit. coreos-metadata will fetch information about the running instance, and write it to a file. The service we want to run can then source this file to get the information when it's starting up.
+
+As of version 0.2.0, ct has support for making this easy for users. In specific sections of a config, users can enter in dynamic data between `{}`, and ct will handle enabling the coreos-metadata service and using the information it provides.
+
+The available information varies by provider, and is expressed in different variables by coreos-metadata, so if this feature is used a `--provider` flag must be passed to ct. Currently, the `etcd` and `flannel` sections are the only ones supporting this.
+
+[coreos-metadata]: https://github.com/coreos/coreos-metadata/
+
+## Supported data by provider
+
+This is the information available in each provider.
```

Comment:
We still want to include some form of it here I think. The names ct uses to refer to each field is different than coreos-metadata, because coreos-metadata's names vary across cloud providers.